### DISCUSSION TITLE
Update trace cable schema URL with new domain name ?

### DISCUSSION THREAD

**Discussion by gfillol** [netbox-community/netbox#16897] [Help Wanted!] [answered]
Hi, at some point we have changed our domain name to access NetBox, and NetBox works fine using the new domain name.

Today with v4.0.7 we started using the trace cable page, and we noticed that the URL in schema is created with the OLD domain name.

Where can we configure the new domain name to update the URL in the trace cable page schema ?

Regards

![2024-07-12_11-27](https://github.com/user-attachments/assets/fe9e657a-e8dc-4550-8a9d-66d2c2eb705a)

**Comment by candlerb** [comment, marked as answer]
Netbox builds absolute URLs for images, and it gets the hostname from the incoming request's Host: or X-Forwarded-Host: header.

Maybe you have a proxy in front of Netbox that's still passing the old name.  Also check your Apache or Nginx config and compare it against those under `/opt/netbox/contrib/`. e.g. if you're using Nginx then check your line `proxy_set_header X-Forwarded-Host`, which should be

```
proxy_set_header X-Forwarded-Host $http_host;
```

If you're still not sure where it's coming from, then run tcpdump on the Netbox server:

```
tcpdump -i lo -nn -s0 -A tcp port 8001
```

Hit it with any request (e.g. just the home page is fine), and look at all the request headers. See which one still has the old domain name in it.

**Reply by gfillol** [reply]
Right, it was as simple as a bad apache configuration with a "ServerName" using ancient domain name. Still don't get why NetBox was mostly running fine with a such bad config. But it works all fine now, thanks !

**Reply by candlerb** [reply]
> Still don't get why NetBox was mostly running fine with a such bad config

It's because *most* of the links that Netbox generates are relative URLs, so the browser itself prepends the hostname it was using to access the site.

Images generated by Netbox have absolute URLs. I'm not entirely sure why that is, but it could be so that you can copy-paste an SVG diagram into another system (a documentation wiki, say) and the links remain clickable.

**Reply by gfillol** [reply]
Yep relatives URL are no problem.
But the incoming packets should not have be "catch" by apache to serve NetBox, du to bad "ServerName" config.
Now I get it : the NetBox apache conf is the first and default one. NetBox is served wherever the domain name is, but others app domains name.

**Reply by candlerb** [reply]
I agree, you've not provided a good explanation why changing `ServerName` had an effect. But without seeing your proxy configuration, it's hard to say.

Maybe you're missing `ProxyPreserveHost On` from the [config](https://github.com/netbox-community/netbox/blob/v4.0.7/contrib/apache.conf#L11).
